### DISCUSSION TITLE
HIP-13: Hedera Name Service

### DISCUSSION THREAD

**Discussion by kenthejr** [hiero-ledger/hiero-improvement-proposals#56] [Edits]
This discussion is designated as the discussions-to target of [hip-13](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-13.md). Use this thread to further develop the HIP and move it through to its final lifecycle state.

**Comment by H-Bart** [comment]
I am proposing the following amendments to the HIP that would make HNS a more direct competitor to existing decentralised domain services and in particular [Handshake](https://handshake.org/):

1. The Hedera Council should blacklist [all existing TLDs registered with ICANN](https://data.iana.org/TLD/tlds-alpha-by-domain.txt)
2. Instead of auctioning off SLDs under the TLD `.hh` all TLDs other than those blacklisted can be auctioned off by the Hedera network in a procedure similar to that already described.
3. To facilitate a more gradual release of TLDs into the market(in addition to the already proposed hard cap of active TLD auctions) the mechanism described below can be implemented:
    * A TLD may be registered if `hash(TLD)%N ≥ i`, where `N` is the time before all TLDs are registrable and `i`(0-indexed) is the discrete unit of time(measured in the same unit as `N`) that has elapsed since TLD registration started.
    * Take for example the TLD `.hbar`:
```python
TLD = '.hbar'
N = 52 # measured in weeks
available_from = hash(TLD)%N # measured in the same unit as N(in this case weeks)
# returns 8, therefore the TLD ".hbar" can be registered starting from week 8(0-indexed) OR week 9(1-indexed)
```
4. TLD owners can register SLDs under their respective TLDs. These SLDs are not tied in ownership to a TLD in the same way subdomains are tied to an SLD(as described in the existing HIP), hence SLDs can be registered by a TLD owner but ownership of an SLD is not tied to its TLD owner, this allows for an entire marketplace whereby TLD owners can register and sell/auction off SLDs.
5. The Hedera Council should whitelist the SLDs for the top 100,000 sites listed on [Alexa](https://www.alexa.com/) and reserve their corresponding TLDs. Whitelisted domain owners can be onboarded onto HNS as previously described and claim their reserved TLDs. For example the TLD `.hedera` will be reserved for the domain owner of `hedera.com` who on proving ownership of the aforementioned domain to the Hedera network will be afforded ownership of the TLD `.hedera` on HNS.
